pkg/service/rtcservice.go

```diff
@@ -72,11 +72,20 @@ func (s *RTCService) validate(r *http.Request) (string, routing.ParticipantInit,
 	roomName := r.FormValue("room")
 	reconnectParam := r.FormValue("reconnect")
 	autoSubParam := r.FormValue("auto_subscribe")
+	publishParam := r.FormValue("publish")
 
 	if onlyName != "" {
 		roomName = onlyName
 	}
 
+	// this is new connection for existing participant -  with publish only permissions
+	if publishParam != "" {
+		// Make sure grant does not have 'subscribe' set,
+		claims.Video.SetCanSubscribe(false)
+		claims.Video.SetCanPublish(true)
```

Comment:
so for security, I don't think we should set publish to true. it should instead be a check, if publish is not set to true (or nil), then should reject connection.